services/ai-engine/main.py

```
@@ -0,0 +1,96 @@
+        raise HTTPException(status_code=400, detail="Unsupported file type. Only PDF is allowed.")
+    
+    content = await file.read()
+    if len(content) > MAX_FILE_SIZE:
+        raise HTTPException(status_code=413, detail="File too large. Maximum size is 10MB.")
+        
+    if len(content) == 0:
+        raise HTTPException(status_code=400, detail="Empty file uploaded.")
+        
+    try:
+        doc = fitz.open(stream=content, filetype="pdf")
+    except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)